lib/workers/testbot.js

```diff
@@ -0,0 +1,182 @@
+    this.board.serialWrite(HW_SERIAL5, [ command, a_, b_ ])
+  }
+
+  async connectSdToTarget (sw) {
+    this.sendCommand(SD_RESET_ENABLE)
+    await Bluebird.delay(10)
+    this.sendCommand(SD_RESET_DISABLE)
+
+    if (sw) {
+      this.board.digitalWrite(LED_PIN, this.board.LOW)
+      this.board.digitalWrite(SD_MUX_SEL_PIN, this.board.LOW)
+    } else {
+      this.board.digitalWrite(LED_PIN, this.board.HIGH)
+      this.board.digitalWrite(SD_MUX_SEL_PIN, this.board.HIGH)
+    }
+
+    return Bluebird.delay(2000)
+  }
+
+  powerTarget (sw) {
+    sw ? this.sendCommand(ENABLE_VOUT_SW) : this.sendCommand(DISABLE_VOUT_SW)
+  }
+
+  async flash (image) {
+    await this.connectSdToTarget(false)
+
+    // For linux, udev will provide us with a nice id for the testbot
+    const destination = await getDevInterface(this.options.devicePath)
+
+    const drives = await drivelist.listAsync()
+    const drive = _.find(drives, {
+      device: destination
+    })
+
+    if (!drive) {
+      throw new Error(`Cannot find ${destination}`)
+    }
+
+    await mountutils.unmountDiskAsync(drive.device)
+    const driveFileDescriptor = await fs.openAsync(drive.raw, 'rs+')
+    const imageSize = await fs.statAsync(image).get('size')
+
+    const emitter = imageWrite.write({
+      fd: driveFileDescriptor,
+      device: drive.raw,
+      size: drive.size
+    }, {
+      stream: fs.createReadStream(image),
+      size: imageSize
+    }, {
+      check: true
+    })
+
+    if (!process.env.CI) {
+      const bar = new visuals.Progress(`Flashing ${destination}`)
+      emitter.on('progress', (data) => {
+        bar.update({
+          percentage: data.percentage,
+          eta: data.eta
+        })
+      })
+    }
+
+    await new Bluebird((resolve, reject) => {
+      emitter.once('error', reject)
+      emitter.once('done', () => {
+        resolve(drive.device)
+      })
+    })
+
+    await fs.closeAsync(driveFileDescriptor).delay(2000)
+    await mountutils.unmountDiskAsync(drive.device)
+  }
+
+  async ready () {
+    await new Bluebird((resolve, reject) => {
+      this.board.once('ready', resolve)
+      this.board.once('error', reject)
+    })
+
+    this.board.pinMode(LED_PIN, this.board.MODES.OUTPUT)
+
+    // Power managment configuration
+    this.sendCommand(ENABLE_FAULTRST)
+    await Bluebird.delay(1000)
+    this.sendCommand(WRITE_DAC_REG, 5)
+    await Bluebird.delay(1000)
+    this.sendCommand(ENABLE_VREG)
+    await Bluebird.delay(1000)
+
+    // SD card managment configuration
+    this.board.pinMode(SD_MUX_SEL_PIN, this.board.MODES.OUTPUT)
+    this.board.digitalWrite(SD_MUX_SEL_PIN, this.board.LOW)
+    this.board.pinMode(USB_MUX_SEL_PIN, this.board.MODES.OUTPUT)
+    this.board.digitalWrite(USB_MUX_SEL_PIN, this.board.LOW)
+  }
+
+  async on () {
+    console.log('Switching SD card to device...')
+    await this.connectSdToTarget(true)
+
+    console.log('Switching testbot on...')
+    this.powerTarget(true)
```

Comment:
The booleans here look really weird. Maybe we should also have `powerTarget` and `shutdownTarget` or something like that?